amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeServiceImpl.java

```diff
@@ -0,0 +1,213 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.streamnative.pulsar.handlers.amqp;
+
+import static org.apache.qpid.server.protocol.ErrorCodes.INTERNAL_ERROR;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.service.persistent.PersistentTopic;
+import org.apache.qpid.server.exchange.ExchangeDefaults;
+import org.apache.qpid.server.protocol.ErrorCodes;
+import org.apache.qpid.server.protocol.v0_8.AMQShortString;
+import org.apache.qpid.server.protocol.v0_8.FieldTable;
+import org.apache.qpid.server.protocol.v0_8.transport.AMQMethodBody;
+import org.apache.qpid.server.protocol.v0_8.transport.ExchangeBoundOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.ExchangeDeleteOkBody;
+import org.apache.qpid.server.protocol.v0_8.transport.MethodRegistry;
+
+/**
+ * Logic of exchange.
+ */
+@Slf4j
+public class ExchangeServiceImpl implements ExchangeService {
+    private ExchangeContainer exchangeContainer;
+
+    public ExchangeServiceImpl(ExchangeContainer exchangeContainer) {
+        this.exchangeContainer = exchangeContainer;
+    }
+
+    private void handleDefaultExchangeInExchangeDeclare(AmqpChannel channel, AMQShortString exchange) {
+        if (isDefaultExchange(exchange)) {
+            StringBuffer sb = new StringBuffer();
+            sb.append("Attempt to redeclare default exchange: of type ")
+                    .append(ExchangeDefaults.DIRECT_EXCHANGE_CLASS);
+            channel.closeChannel(ErrorCodes.ACCESS_REFUSED, sb.toString());
+        }
+    }
+
+    private String formatString(String s) {
+        return s.replaceAll("\r", "").
+                replaceAll("\n", "").trim();
+    }
+
+
+    public void exchangeDeclare(AmqpChannel channel, AMQShortString exchange, AMQShortString type,
+                                boolean passive, boolean durable, boolean autoDelete,
+                                boolean internal, boolean nowait, FieldTable arguments) {
+        int channelId = channel.getChannelId();
+        AmqpConnection connection = channel.getConnection();
+        if (log.isDebugEnabled()) {
+            log.debug("RECV[{}] ExchangeDeclare[ exchange: {},"
+                            + " type: {}, passive: {}, durable: {}, autoDelete: {}, internal: {}, "
+                            + "nowait: {}, arguments: {} ]", channelId, exchange,
+                    type, passive, durable, autoDelete, internal, nowait, arguments);
+        }
+
+        handleDefaultExchangeInExchangeDeclare(channel, exchange);
+
+        String exchangeName = formatString(exchange.toString());
+        final MethodRegistry methodRegistry = connection.getMethodRegistry();
```

Comment:
NIT: Since this is only used on the next line, you probably don't need to pull it out into a variable.

Just use this:

`final AMQMethodBody declareOkBody = connection.getMethodRegistry().createExchangeDeclareOkBody();`